OpenSim/Simulation/StatesTrajectory.h

```diff
@@ -0,0 +1,641 @@
+ *     ...
+ * }
+ * @endcode
+ * @relates IteratorRange
+ */
+template <class Iterator>
+IteratorRange<Iterator> makeIteratorRange(Iterator first, Iterator last) {
+    return IteratorRange<Iterator>(first, last);
+}
+
+
+class Storage;
+class Model;
+
+
+// Design note: This class is part of OpenSim instead of Simbody since Simbody
+// users are likely to be interested in a more general State container that
+// doesn't assume the states are sequential in time.
+
+/** A sequence of SimTK::State%s that can be saved to a plain-text OSTATES
+ * file.  The trajectory can also be populated from such a file. You may obtain
+ * a StatesTrajectory from a simulation, or other numerical methods whose task
+ * is to produce a trajectory of states. Users can modify the trajectory by
+ * appending states to it, but users cannot modify the individual states that
+ * are already in the trajectory.
+ *
+ * This class was introduced in OpenSim version 4.0, and enables scripting
+ * (Python/MATLAB) and C++ users to postprocess their results much more
+ * flexibly than with an Analysis.
+ *
+ * ### Guarantees
+ * This class is designed to ensure the following:
+ * - The states are ordered nondecreasing in time (adjacent states *can* have
+ *   the same time).
+ * - All states in the trajectory are consistent with each other (see
+ *   isConsistent()).
+ *
+ * @note These gaurantees apply when using this class through C++, Java,
+ * or the %OpenSim GUI, but **not** through Python or MATLAB. This is because
+ * Python and MATLAB do not enforce constness and thus allow modifying the
+ * trajectory.
+ *
+ * ### Using with a %Model 
+ * A StatesTrajectory is not very useful on its own, since neither the
+ * trajectory nor the contained states know of the names that the Component%s
+ * give to the state variables. You probably want to use the trajectory with an
+ * OpenSim::Model, through which the state variables have a meaning (e.g.,
+ * `model.getStateVariableValue(states[0], "soleus_r/activation")`).
+ *
+ * SimTK::State%s have a tight association with a specific OpenSim::Model
+ * (actually, with the SimTK::System within an OpenSim::Model). However,
+ * neither the StatesTrajectory nor the OSTATES file knows the model to which
+ * it corresponds. So, for example, you could use a single StatesTrajectory
+ * with a generic gait2392 model as well as with a scaled (subject-specific)
+ * gait2392 model. This flexibility may be beneficial in some scenarios, but
+ * also allows one to accidentally use the wrong model with a given states
+ * trajectory, potentially leading to silent errors that could compromise a
+ * scientific study.
+ *
+ * To increase your confidence that a StatesTrajectory matches a given Model,
+ * you can perform some weak checks with isCompatibleWith().
+ *
+ * TODO acceleration-level calculations?
+ *
+ * ### File format
+ * StatesTrajectory files use the file extension `.OSTATES`, with the XML
```

Comment:
Could you clarify if you mean just when extension appears with the period (`.OSTATES`), or even when it appears without the period (`OSTATES`)? 

I'm indfferent in the case that the period is there (`.ostates`), but I still prefer upper-case when leaving out the period (`OSTATES`); perhaps just because it seems more common. Though, I'll follow whatever your preference is.